tests/integration/opt_test.py

```diff
@@ -0,0 +1,211 @@
+        return x
+
+    # Act.
+    cx = execute(main, with_='--x 1')
+
+    # Assert.
+    assert cx == '1'
+
+
+def test_one_opt_with_default_value():
+    # Arrange.
+    @cli
+    def main(x: Opt = '2'):
+        return x
+
+    # Act.
+    cx = execute(main, with_='--x 1')
+
+    # Assert.
+    assert cx == '1'
+
+
+def test_one_opt_with_default_value_but_not_specified_in_cli():
+    # Arrange.
+    @cli
+    def main(x: Opt = '2'):
+        return x
+
+    # Act.
+    cx = execute(main, with_='')
+
+    # Assert.
+    assert cx == '2'
+
+
+def test_one_opt_not_specified_in_cli():
+    # Arrange.
+    @cli
+    def main(x: Opt):
+        return x
+
+    # Act.
+    ex = execute(main, with_='')
+
+    # Assert.
+    assert isinstance(ex, BaseException)
+
+
+def test_one_opt_with_default_value_but_different_opt_specified_in_cli():
+    # Arrange.
+    @cli
+    def main(x: Opt = '2'):
+        return x
+
+    # Act.
+    ex = execute(main, with_='--y 1')
+
+    # Assert.
+    assert isinstance(ex, BaseException)
+
+
+def test_one_opt_without_value():
+    # Arrange.
+    @cli
+    def main(x: Opt):
+        return x
+
+    # Act.
+    ex = execute(main, with_='--x')
+
+    # Assert.
+    assert isinstance(ex, BaseException)
+
+
+def test_two_opts():
+    # Arrange.
+    @cli
+    def main(x: Opt, y: Opt):
+        return x, y
+
+    # Act.
+    cx = execute(main, with_='--x 1 --y 2')
+
+    # Assert.
+    assert cx == ('1', '2')
+
+
+def test_two_opts_in_different_order():
+    # Arrange.
+    @cli
+    def main(x: Opt, y: Opt):
+        return x, y
+
+    # Act.
+    cx = execute(main, with_='--y 2 --x 1')
+
+    # Assert.
+    assert cx == ('1', '2')
+
+
+def test_one_opt_before_flag():
+    # Arrange.
+    @cli
+    def main(x: Opt, y: Flag):
+        return x, y
+
+    # Act.
+    cx = execute(main, with_='--x 1 --y')
+
+    # Assert.
+    assert cx == ('1', True)
+
+
+def test_one_opt_before_arg():
+    # Arrange.
+    @cli
+    def main(x: Opt, y: Arg):
+        return x, y
+
+    # Act.
+    cx = execute(main, with_='--x 1 2')
+
+    # Assert.
+    assert cx == ('1', '2')
+
+
+def test_one_opt_specified_without_value_but_another_flag():
+    # Arrange.
+    @cli
+    def main(x: Opt):
+        return x
+
+    # Act.
+    ex = execute(main, with_='--x --y')
+
+    # Assert.
+    assert isinstance(ex, BaseException)
+
+
+def test_one_opt_with_implicit_short():
+    # Arrange.
+    @cli
+    def main(xyz: Opt):
+        return xyz
+
+    # Act.
+    cx = execute(main, with_='-x 1')
+
+    # Assert.
+    assert cx == '1'
+
+
+def test_one_opt_with_explicit_short():
+    # Arrange.
+    @cli
+    def main(x: Opt(short='y')):
+        return x
+
+    # Act.
+    cx = execute(main, with_='-y 1')
+
+    # Assert.
+    assert cx == '1'
+
+
+def test_opt_specified_twice():
+    # Arrange.
+    @cli
+    def main(x: Opt):
+        return x
+
+    # Act.
+    cx = execute(main, with_='--x 1 --x 2')
+
+    # Assert.
+    assert cx is '2'
```

Comment:
Shouldn't there be `==`?

Reply:
Yeah, fixed. Seems like I'm in love with the `is` notation.

Reply:
Agree, `is` looks much better than `==`, and previously I would use `is` all the time. I was so disappointed when I figured out that it was incorrect to use it for string comparison.